.github/workflows/tests.yml

```
@@ -0,0 +1,70 @@
+        path: test-results/
+
+    - name: Upload coverage results
+      if: always()
+      uses: actions/upload-artifact@v3
+      with:
+        name: coverage-${{ matrix.os }}
+        path: coverage/
+
+    - name: Upload coverage to Codecov
+      if: always()
+      uses: codecov/codecov-action@v3
```

actionlint (1.7.4)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)